fail2ban/client/fail2banregex.py

```diff
@@ -0,0 +1,599 @@
+				total += match
+				if (match or self._verbose):
+					out.append("%2d) [%d] %s" % (cnt+1, match, failregex.getFailRegex()))
+
+				if self._verbose and len(failregex.getIPList()):
+					for ip in failregex.getIPList():
+						timeTuple = time.localtime(ip[2])
+						timeString = time.strftime("%a %b %d %H:%M:%S %Y", timeTuple)
+						out.append(
+							"    %s  %s%s" % (
+								ip[1],
+								timeString,
+								ip[-1] and " (multiple regex matched)" or ""))
+
+			output( "\n%s: %d total" % (title, total) )
+			pprint_list(out, " #) [# of hits] regular expression")
+			return total
+
+		# Print title
+		total = print_failregexes("Failregex", self._failregex)
+		_ = print_failregexes("Ignoreregex", self._ignoreregex)
+
+
+		if self._filter.dateDetector is not None:
+			output( "\nDate template hits:" )
+			out = []
+			for template in self._filter.dateDetector.templates:
+				if self._verbose or template.hits:
+					out.append("[%d] %s" % (
+						template.hits, template.name))
+			pprint_list(out, "[# of hits] date format")
+
+		output( "\nLines: %s" % self._line_stats, )
+		if self._time_elapsed is not None:
+			output( "[processed in %.2f sec]" % self._time_elapsed, )
+		output( "" )
+
+		if self._print_all_matched:
+			self.printLines('matched')
+		if not self._print_no_ignored:
+			self.printLines('ignored')
+		if not self._print_no_missed:
+			self.printLines('missed')
+
+		return True
+
+	def file_lines_gen(self, hdlr):
+		for line in hdlr:
+			yield self.decode_line(line)
+
+	def start(self, opts, args):
+
+		cmd_log, cmd_regex = args[:2]
+
+		if not self.readRegex(cmd_regex, 'fail'):
+			return False
+
+		if len(args) == 3 and not self.readRegex(args[2], 'ignore'):
+			return False
+
+		if os.path.isfile(cmd_log):
+			try:
+				hdlr = open(cmd_log, 'rb')
+				output( "Use         log file : %s" % cmd_log )
+				output( "Use         encoding : %s" % self.encoding )
+				test_lines = self.file_lines_gen(hdlr)
+			except IOError, e:
+				output( e )
+				return False
+		elif cmd_log == "systemd-journal": # pragma: no cover
+			if not journal:
+				output( "Error: systemd library not found. Exiting..." )
+				return False
+			myjournal = journal.Reader(converters={'__CURSOR': lambda x: x})
+			journalmatch = self._journalmatch
+			self.setDatePattern(None)
+			if journalmatch:
+				try:
+					for element in journalmatch:
+						if element == "+":
+							myjournal.add_disjunction()
+						else:
+							myjournal.add_match(element)
+				except ValueError:
+					output( "Error: Invalid journalmatch: %s" % shortstr(" ".join(journalmatch)) )
+					return False
+			output( "Use    journal match : %s" % " ".join(journalmatch) )
+			test_lines = journal_lines_gen(myjournal)
+		else:
+			output( "Use      single line : %s" % shortstr(cmd_log) )
+			test_lines = [ cmd_log ]
+		output( "" )
+
+		self.process(test_lines)
+
+		if not self.printStats():
+			return False
+
+		return True
+
+
+def exec_command_line(): # pragma: no cover
```

Comment:
this was just an initial coverage :) I will make further ...
And you don't need remove a "no cover" statement, because we can see it all grayed in coveralls ...